### PR TITLE
west.yml: update modules/hal/stm32/hal_stm32 for stm32h503

### LINKED ISSUE #65280: STM32H503 - simple board fails compiling
**Describe the bug**

Creating a simple board using a STM32H503 fails. Tested with main line and v.3.5.0 release. The build fails, because there are unknown types in the HAL driver:
- unknown type name 'DCACHE_TypeDef'
- unknown type name 'DLYB_TypeDef'
- unknown type name 'SAI_Block_TypeDef'

**To Reproduce**
1. Navigate to samples/basic/minimal
2. Extract [boards.zip](https://github.com/zephyrproject-rtos/zephyr/files/13375378/boards.zip)
3. west build -b h5test -p -- -DBOARD_ROOT=.

The h563test board is working.

**Expected behavior**
A successful build

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain: Zephyr SDK
 - Version: v3.5.0

### PR DESCRIPTION
**Pull request by FRASTM**
Update the stm32h5xx/drivers/include/
to enable the SAI, DelayBlock and DCACHE only if the instance exists in the soc. 
Which is not the case for stm32h503x device 

https://github.com/zephyrproject-rtos/hal_stm32/pull/181/

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/65280

This will avoid compilation issue when building a stm32h503 board:
- `error: unknown type name 'DCACHE_TypeDef'`
- `error: unknown type name 'DLYB_TypeDef'`
- `error: unknown type name 'SAI_Block_TypeDef'`

### CODE REVIEW

**Comment by FRASTM** [comment]
do not merge until https://github.com/zephyrproject-rtos/hal_stm32/pull/181 is merged

**Comment by FRASTM** [comment]
rebase after merge of the module/hal/stm32cube revision: pull/181/head

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**
